library/std/src/sys/windows/net.rs

```diff
@@ -159,7 +159,7 @@ impl Socket {
                 }
 
                 let mut timeout = c::timeval {
-                    tv_sec: timeout.as_secs() as c_long,
+                    tv_sec: cmp::min(timeout.as_secs(), libc::time_t::MAX as u64) as c_long,
```

Comment:
Hm, this should be using `c_long::MAX` to match `tv_sec`'s type.

Reply:
Updated